grapher/axis/Axis.ts

```diff
@@ -462,32 +463,28 @@ export class HorizontalAxis extends AbstractAxis {
 
         // Make sure the start and end values are present, if they're whole numbers
         const startEndPrio = this.scaleType === ScaleType.log ? 2 : 1
-        if (
-            domain[0] % 1 === 0 &&
-            // Make sure that start value is not already present.
-            !ticks.some((tick) => tick.value === domain[0])
-        )
+        if (domain[0] % 1 === 0)
             ticks = [
                 {
                     value: domain[0],
                     priority: startEndPrio,
                 },
                 ...ticks,
             ]
-        if (
-            domain[1] % 1 === 0 &&
-            this.hideFractionalTicks &&
-            // Make sure that end value is not already present.
-            !ticks.some((tick) => tick.value === domain[1])
-        )
+        if (domain[1] % 1 === 0 && this.hideFractionalTicks)
             ticks = [
                 ...ticks,
                 {
                     value: domain[1],
                     priority: startEndPrio,
                 },
             ]
-        return uniq(ticks)
+
+        // sort by value, then priority.
+        // this way, we don't end up with two ticks of the same value but different priorities.
+        // instead, we deduplicate by choosing the highest priority (i.e. lowest priority value).
+        const sortedTicks = sortBy(ticks, (tick) => [tick.value, tick.priority])
```

Comment:
I think we need to use [`orderBy`](https://lodash.com/docs/4.17.15#orderBy) because `sortBy` seems to compare arrays by their string value?

```ts
_.sortBy([[100, 1], [30, 1], [1, 1]], (d) => d) 
// [
//   [1, 1]
//   [100, 1]
//   [30, 1]
// ]
```

I _think_ this should work:

```suggestion
        const sortedTicks = orderBy(ticks, [(t) => t.value, (t) => t.priority])
```

Reply:
Ooh, that is a very good catch! Thank you.